tools/syz-declextract/run.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2017 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package main
+
+import (
+	"encoding/json"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/google/syzkaller/pkg/tool"
+)
+
+type compileCommand struct {
+	Arguments []string
+	Directory string
+	File      string
+	Output    string
+}
+
+func run(file string, out chan []byte) {
+	if !strings.HasSuffix(file, ".c") {
+		out <- nil
+		return
+	}
+
+	cmd := exec.Command("syz-declextract", "-p", os.Args[1], file)
+	stdout, err := cmd.Output()
+	if err != nil {
+		tool.Fail(err)
```

Comment:
We should output stderr, otherwise if parsing fails, it won't be possible to understand what went wrong, I think it will just say "process failed with code 1".
This ExitError.Stderr should contain the output:
https://pkg.go.dev/os/exec#ExitError
Also if it does not include the file name, we should print it as well.

Try to introduce some syntax error into one of kernel source files, and check what this tool prints for it and if it's possible to understand what happened.

Reply:
Done!